src/pages/EmployeeOverview/components/FilterEmployeeModal.jsx

```diff
@@ -0,0 +1,175 @@
+ */
+function FilterEmployeeModal(props) {
+  const theme = useTheme();
+
+  //Styling for the multiple select, to highlight the selected items
+  function getStyles(array, item, theme) {
+    return {
+      fontWeight:
+        array.indexOf(item) === -1
+          ? theme.typography.fontWeightRegular
+          : theme.typography.fontWeightMedium,
+    };
+  }
+
+  const [selectedJobTitle, setSelectedJobTitle] = useState([]);
+
+  const uniqueJobTitles = [
+    ...new Set(props.employees?.map((employee) => employee.jobTitle)),
+  ];
+
+  /**
+   * The function `handleJobTitleChange` takes an event object and updates the selected job titles based
+   * on the value of the event target.
+   */
+  const handleJobTitleChange = (event) => {
+    const {
+      target: { value },
+    } = event;
+    setSelectedJobTitle(typeof value === "string" ? value.split(",") : value);
+  };
+
+  const [selectedCourse, setSelectedCourse] = useState([]);
+  const uniqueCourseNamesArray = [
+    ...new Set(
+      props.employees?.flatMap((employee) =>
+        employee.courses?.map((course) => course.name)
+      )
+    ),
+  ];
+  /**
+   * The function `handleCourseChange` takes an event object and updates the `selectedCourse` state based
+   * on the value of the event target.
+   */
+
+  const handleCourseChange = (event) => {
+    const {
+      target: { value },
+    } = event;
+    setSelectedCourse(typeof value === "string" ? value.split(",") : value);
+  };
+
+  function applyFilter() {
+    props.handleClose({
+      jobTitles: selectedJobTitle || [],
+      courses: selectedCourse || [],
+    });
+  }
+
+  return (
+    <Dialog open={props.open} onClose={props.handleClose}>
+      <DialogTitle>Filter your Employees</DialogTitle>
+      <DialogContent>
+        <DialogContentText>Filter by Job Titles</DialogContentText>
+        <FormControl sx={{ m: 1, width: 300 }}>
+          <InputLabel id="job-title-select-label">Job Title</InputLabel>
+          <Select
+            labelId="job-title-select-label"
+            id="job-title-select"
+            multiple
+            value={selectedJobTitle}
+            onChange={handleJobTitleChange}
+            input={<OutlinedInput label="JobTitle" />}
+            MenuProps={MenuProps}
+          >
+            {uniqueJobTitles.map((jobTitle) => (
+              <MenuItem
+                key={jobTitle}
+                value={jobTitle}
+                style={getStyles(selectedJobTitle, jobTitle, theme)}
+              >
+                {jobTitle}
+              </MenuItem>
+            ))}
+          </Select>
+        </FormControl>
+        <DialogContentText>Filter by Course Name</DialogContentText>
+        <FormControl sx={{ m: 1, width: 300 }}>
+          <InputLabel id="course-name-select-label">Courses</InputLabel>
+          <Select
+            labelId="course-name-select-label"
+            id="course-name-select"
```

Comment:
Why this ids?

Reply:
I copied this from the documentation.
They recommend having one

Reply:
But why? I think that would be our first id and it is never used, or am I mistaken?